pkg/controller/jobframework/reconciler.go

```diff
@@ -437,6 +437,7 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 					workload.SetRequeuedCondition(wl, evCond.Reason, evCond.Message, true)
 				}
 				_ = workload.UnsetQuotaReservationWithCondition(wl, "Pending", evCond.Message)
+				r.record.Event(wl, corev1.EventTypeNormal, kueue.WorkloadEvicted, evCond.Message)
```

Comment:
The issue description https://github.com/kubernetes-sigs/kueue/issues/2012 indicates we are already getting an event Preempted for preemptions. So, I guess we will not have: Evicted and Preempted, which is a little redundant, but maybe ok. 

@mbobrovskyi please check what happens in case of preemptions and post in the PR description as you did for the PodsReady timeouts.

Reply:
After further analysis it seems to be the case, I can see the options:
1. Introduce generic Evicted event (as currently), along with the existing Preempted event
2. Send the scenario-specific event (as currently for preemptions)
3. do 1 and 2

Maybe for now we can start with 2, just to make the handling symmetric. We could wrap sending the event and bumping the metric into one function `ReportEvictedWorkload`: https://github.com/kubernetes-sigs/kueue/blob/baefcc433a1f9fd844f91a6c12327f163838c7f6/pkg/scheduler/preemption/preemption.go#L190-L191. 

Any view @alculquicondor @tenzen-y ?

Reply:
> @mbobrovskyi please check what happens in case of preemptions and post in the PR description as you did for the PodsReady timeouts.

```bash
Events:
  Type    Reason         Age                From                        Message
  ----    ------         ----               ----                        -------
  Normal  QuotaReserved  68s (x2 over 68s)  kueue-admission             Quota reserved in ClusterQueue cluster-queue, wait time since queued was 1s
  Normal  Admitted       68s (x2 over 68s)  kueue-admission             Admitted by ClusterQueue cluster-queue, wait time since reservation was 0s
  Normal  Preempted      68s                kueue-admission             Preempted to accommodate a workload (UID: 39043e28-4d9e-408c-bb84-2f6b69b0fc91) in the ClusterQueue
  Normal  Evicted        68s                batch/job-kueue-controller  Preempted to accommodate a workload (UID: 39043e28-4d9e-408c-bb84-2f6b69b0fc91) in the ClusterQueue
```

Also added to description.

Reply:
Next to all calls to `metrics.ReportEvictedWorkloads` makes sense, so that we can be more specific.

Reply:
Thanks for the update.

While I think both generic "Evicted" event, and scenario-specific might be useful, since we already have the scenario-specific Preempted, I think adding analogous events for other scenarios is enough, and would satisfy  the original motivation for the issue.

Maybe we can add the generic one later, if we have an ask from a user.

Reply:
TBH, both approaches ( 1 and 2 ) should work fine. But if we introduce the generic Eviction events, we need to take care of the order of events (Preemppted -> Evicted vs Evicted -> Preempted ).

Anyway, events existing would be worth it rather than no events.

Reply:
So, I'm thinking maybe we could reuse the naming pattern `EvictedDueTo...`, as we already do [here](https://github.com/kubernetes-sigs/kueue/blob/a7dcb6da5572bcbb403485051095f1ee8ace56ec/pkg/controller/jobframework/reconciler.go#L837). It would be both indicating that the workload is evicted, and be specific. 

Renaming `Preempted` to `EvictedDueToPreempted` might be a downside, but seems acceptable. WDYT @alculquicondor @tenzen-y ?

Reply:
because the `Preempted` event has been out there for a while, I would be worried about changing the constant.

Reply:
That's right, since 0.3.0. In that case, we could keep `Preempted` for the preemption scenario, and
1. add `EvictedDueTo` events for all scenarios
2. add `EvictedDueTo` events for all other scenarios

Maybe with (1.) we could gradually deprecate and remove "Preempted", for example, when moving to v1?

Reply:
> add EvictedDueTo events for all other scenarios

To do not duplicate events, I think it's enough.

Reply:
@alculquicondor WDYT?